spec/std/yaml/from_yaml_spec.cr

```diff
@@ -0,0 +1,359 @@
+require "spec"
+require "yaml"
+
+private struct ObjectStub
+  getter called : Bool
+
+  def self.new(pull : YAML::PullParser)
+    pull.read_scalar
+    new(true)
+  end
+
+  def initialize(@called)
+  end
+end
+
+describe "Object.from_yaml" do
+  it "should create call .new on with a YAML::PullParser" do
+    ObjectStub.from_yaml("hello").called.should be_true
+  end
+end
+
+describe "Nil.from_yaml" do
+  it "should serialize and deserialize" do
+    Nil.from_yaml(nil.to_yaml).should eq nil
+  end
+
+  values = {"", "NULL", "Null", "null", "~"}
+
+  values.each do |value|
+    it %(zshould return nil if "#{value}") do
+      # Test will an array since a standalone empty value shows as STREAM_END
+      Array(Nil).from_yaml("- " + value).should eq [nil]
+    end
+  end
+
+  values.each do |value|
+    it %(should raise if "#{value}" is quoted with double quotes) do
+      expect_raises(YAML::ParseException) do
+        Nil.from_yaml %("#{value}")
+      end
+    end
+
+    it %(should raise if "#{value}" is quoted with single quotes) do
+      expect_raises(YAML::ParseException) do
+        Nil.from_yaml "'#{value}'"
+      end
+    end
+
+    it "should raise if not a null value" do
+      expect_raises(YAML::ParseException) do
+        Nil.from_yaml "hello"
+      end
+    end
+  end
+
+  describe "Bool.from_yaml" do
+    describe "true values" do
+      it "should serialize and deserialize" do
+        Bool.from_yaml(true.to_yaml).should eq true
+      end
+
+      values = {"true", "True", "TRUE", "on", "On", "ON", "y", "Y", "yes", "Yes", "YES"}
```

Comment:
Why is this here again? I thought we agreed on inlining the values instead of having dozens of duplicate specs?

Reply:
Probably because I brought up a counter-argument [here](https://github.com/crystal-lang/crystal/pull/4838#discussion_r134877472) (I meant it more as a question/discussion)

Reply:
A discussion would probably be more appropriate then jumping back and forth... This change makes the previous comments disappear... (hide from plain sight).

I don't think your argument is valid, because specs' goal is primarily to identify if failures exists in the code, not necessarily exactly why and where. This would probably need some further exploration anyway. So testing all in one spec is equally safe as individual specs.
In this particular case, it should be quite sufficient to test if all values work: In the unlikely case of a failure, all values will probably be affected equally.
Having individual specs in this place has as good as no real benefit but reduces performance on each and every spec run. That's not worth it.
It also spams spec reports with dozens of successful results which mean nothing since they are all testing exactly the same behaviour.